mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -285,16 +288,33 @@ private void commitTable(FileIO io, ExecutorService executor, JobContext jobCont
 
     Collection<DataFile> dataFiles = dataFiles(executor, location, jobContext, io, true);
 
-    if (dataFiles.size() > 0) {
+    boolean isOverwrite = conf.getBoolean(ConfigProperties.IS_OVERWRITE, false);
+
+    if (isOverwrite) {
+      if (!dataFiles.isEmpty()) {
+        ReplacePartitions overwrite = table.newReplacePartitions();
+        dataFiles.forEach(overwrite::addFile);
+        overwrite.commit();
+        LOG.info("Overwrite commit took {} ms for table: {} with {} file(s)", System.currentTimeMillis() - startTime,
+          table, dataFiles.size());
+      } else if (table.spec().isUnpartitioned()) {
+        table.newDelete().deleteFromRowFilter(Expressions.alwaysTrue()).commit();
+        LOG.info("Cleared table contents as part of empty overwrite for unpartitioned table. " +
+          "Commit took {} ms for table: {}", System.currentTimeMillis() - startTime, table);
+      }
```

Comment:
What is the behavior of using empty to overwrite a partitioned table? Should we print some log?

Reply:
Not do anything, I added some log.